pkg/operation/care/checker.go

```diff
@@ -466,9 +466,10 @@ func computeRequiredControlPlaneDeployments(shoot *gardencorev1beta1.Shoot) (set
 			}
 		}
 
-		if features.DefaultFeatureGate.Enabled(features.MachineControllerManagerDeployment) {
-			requiredControlPlaneDeployments.Insert(v1beta1constants.DeploymentNameMachineControllerManager)
-		}
+		// TODO(rfranzke): Uncomment this code once the MachineControllerManagerDeployment feature gate gets removed.
```

Comment:
This would mean that unhealthy MCM deployment won't be reported until we tackle the TODO.

I was wondering whether we can use a new selector in https://github.com/gardener/gardener/blob/086280a89b5ed79e28a948e626c8d5815e221731/pkg/operation/care/checker.go#L507-L510 and select all the Deployments with `gardener.cloud/role=controlplane` and the MCM deployment (`role=machine-controller-manager`). Then, we won't have false positive failures and we wouldn't need to comment the health check.

Reply:
The `checkControlPlane` function will still check the health of the MCM deployment even if it's not in the list of "required deployments". The only thing that will change with this PR is that a missing MCM will no longer be reported - but in this case, we will probably notice it because of other issues/failing conditions. I think we can live with this for a few releases (the plan is to promote and remove the feature gate relatively soon now that all provider extensions support it).

Reply:
Ah, true that. Thanks for explaining!